torch/csrc/jit/passes/onnx.cpp

```diff
@@ -249,13 +271,33 @@ void NodeToONNX(
         //
         // If onnx shape inference is turned on, the new outputs will have
         // types inferred, and they will be merged with the old types.
-        outputs[i]->setType(MergeInferredType(old->type(), outputs[i]->type()));
+        if (enable_const_folding &&
+            outputs[i]->node()->kind() != c10::onnx::Constant &&
+            ConstantValueMap::HasValue(outputs[i]->debugName())) {
+          // Perform const folding if the node output value is in ConstantValueMap.
```

Comment:
nit: maybe rephrase this comment to be more elaborative. For once I thought it will perform constant folding like compute the actual value, but then realized the value was already computed, and are just fetched.

Reply:
Create a const node if the node output value is in ConstantValueMap.